node/forest_libp2p/src/service.rs

```diff
@@ -180,8 +236,12 @@ impl Libp2pService {
                         NetworkMessage::PubsubMessage { topic, message } => {
                             swarm_stream.get_mut().publish(&topic, message);
                         }
-                        NetworkMessage::RPC { peer_id, request, id } => {
-                            swarm_stream.get_mut().send_rpc_request(&peer_id, request, id);
+                        NetworkMessage::HelloRequest { peer_id, request, id } => {
+                            swarm_stream.get_mut().send_rpc_request(&peer_id, RPCRequest::Hello(request), id);
+                        }
+                        NetworkMessage::BlockSyncRequest { peer_id, request, id, response_channel } => {
+                            self.bs_request_table.lock().await.insert(id, response_channel);
+                            swarm_stream.get_mut().send_rpc_request(&peer_id, RPCRequest::BlockSync(request), id);
```

Comment:
shouldn't need to pass in the id anymore, `send_rpc_request` should change to not use the forked method and let the RequestResponse service handle the id generation and then just insert into the table with the returned ID